node/src/database/db_migrations/migrator_utils.rs

```diff
@@ -0,0 +1,456 @@
+        let db_path = dir_path.join("test_database.db");
+        let connection = Connection::open(&db_path).unwrap();
+        connection
+            .execute(
+                "CREATE TABLE test (
+            name TEXT,
+            count integer
+        )",
+                [],
+            )
+            .unwrap();
+        let correct_statement_1 = "INSERT INTO test (name,count) VALUES ('mushrooms',270)";
+        let erroneous_statement_1 =
+            "INSERT INTO botanic_garden (name, count) VALUES (sunflowers, 100)";
+        let erroneous_statement_2 = "INSERT INTO milky_way (star) VALUES (just_discovered)";
+        let set_of_sql_statements: &[&dyn StatementObject] = &[
+            &correct_statement_1,
+            &erroneous_statement_1,
+            &erroneous_statement_2,
+        ];
+        let mut connection_wrapper = ConnectionWrapperReal::new(connection);
+        let config = DBMigratorInnerConfiguration::new();
+        let external_parameters = make_external_data();
+        let subject = DBMigrationUtilitiesReal::new(&mut connection_wrapper, config).unwrap();
+
+        let result = subject
+            .make_mig_declaration_utils(&external_parameters, &Logger::new("test logger"))
+            .execute_upon_transaction(set_of_sql_statements);
+
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "no such table: botanic_garden"
+        );
+        let connection = Connection::open(&db_path).unwrap();
+        //when an error occurs, the underlying transaction gets rolled back, and we cannot see any changes to the database
+        let assertion: Option<(String, String)> = connection
+            .query_row("SELECT count FROM test WHERE name='mushrooms'", [], |row| {
+                Ok((row.get(0).unwrap(), row.get(1).unwrap()))
+            })
+            .optional()
+            .unwrap();
+        assert!(assertion.is_none()) //means no result for this query
+    }
+
+    #[test]
+    fn execute_upon_transaction_handles_also_statements_that_return_something() {
+        let dir_path = ensure_node_home_directory_exists(
+            "db_migrations",
+            "execute_upon_transaction_handles_also_statements_that_return_something",
+        );
+        let db_path = dir_path.join("test_database.db");
+        let connection = Connection::open(&db_path).unwrap();
+        connection
+            .execute(
+                "CREATE TABLE botanic_garden (
+            name TEXT,
+            count integer
+        )",
+                [],
+            )
+            .unwrap();
+        let statement_1 = "INSERT INTO botanic_garden (name,count) VALUES ('sun_flowers', 100)";
+        let statement_2 = "ALTER TABLE botanic_garden RENAME TO just_garden"; //this statement returns an overview of the new table on its execution
```

Comment:
I'm pretty sure this comment is wrong. Talk to @bertllll and make sure this test tests what it thinks it's testing.